src/interpreter/mod.rs

```diff
@@ -1169,58 +1222,89 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 self.stack[frame].set_local(local, Value::ByVal(val));
                 Ok(())
             }
+            Lvalue::Static(cid) => {
+                let static_val = self.statics.get_mut(&cid).expect("static not cached");
+                assert!(static_val.mutable);
```

Comment:
I haven't been able to produce an example. There's no safe or unsafe way to turn an immutable static into a mutable static without miri allocating (since we need to cast a reference). The reacheable cases report the usual errors.